src/include/primer/hyperloglog_presto.h

```diff
@@ -54,7 +58,20 @@ class HyperLogLogPresto {
    *
    * @returns hash value
    */
-  inline auto CalculateHash(T val) -> hash_t { return std::hash<T>{}(val); }
+  inline auto CalculateHash(T val) -> hash_t {
+    if (std::is_same<T, std::string>::value) {
+      return hash_fn_.GetHash(val);
+    }
+    /** @brief Convert into output stream */
+    std::ostringstream o_stream;
+
+    o_stream << val;  // output stream
+
+    /** @brief Output stream result. */
+    std::string res = o_stream.str();
+
+    return static_cast<hash_t>(std::stoul(res));
+  }
```

Comment:
I remember in the write-up, students are supposed to distinguish between `int` and `std::string` in the basic hyperloglog rather than the presto one. Did you change it?

Reply:
Yes, changing the input type changes, where the string has to be converted to a c_str and then passed on to the function. Hence, removed that for now. Will update it in the writeup.